### ISSUE TITLE
`@typescript-eslint/no-use-before-define` reports `$props` unepectedly

### ISSUE THREAD

**Issue by baseballyama** [sveltejs/eslint-plugin-svelte#1158]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.

### What version of ESLint are you using?

9.23.0

### What version of `eslint-plugin-svelte` are you using?

3.3.3

### What did you do?

Just run ESLint.

### What did you expect to happen?

I don't get lint error

### What actually happened?

<img width="955" alt="Image" src="https://github.com/user-attachments/assets/f62ed6dd-7eb3-4e57-aa5c-cbfb802328f9" />

### Link to **GitHub Repo** with Minimal Reproducible Example

```svelte
<script lang="ts">
  interface Props {
    bar: string;
  }

  const props: Props = $props();
  const state = $state(props);

  console.warn(props.bar);
</script>
```

### Additional comments

It’s caused by conflicting with the store detection logic. I think a fix on the parser side is needed.
We should also check how Svelte itself handles this.
